inspirehep/utils/tickets.py

```diff
@@ -24,26 +24,234 @@
+                  subject="No Subject",
+                  recid=None,
+                  **kwargs):
+    """Creates new RT ticket and returns new ticket id.
+
+    :param queue: where the ticket will be created
+    :type queue: string
+
+    :param requestors: username to set to requestors field of the ticket
+    :type requestors: string
+
+    :param body: message body of the ticket
+    :type body: string
+
+    :param subject: subject of the ticket
+
+    :param recid: record id to be set custom RecordID field
+    :type recid: integer
+
+    :param kwargs: Other arguments possible to set:
+
+                   Cc, AdminCc, Owner, Status,Priority, InitialPriority,
+                   FinalPriority, TimeEstimated, Starts, Due,
+                   ... (according to RT fields)
+
+                   Custom fields CF.{<CustomFieldName>} could be set
+                   with keywords CF_CustomFieldName.
+    :returns: ID of the new ticket or ``-1``, if it fails
+    :rtype: integer
+    """
+    body = _strip_lines(body)
+    # to workaround UnicodeEncodeError raised by rt because of special chars
+    body = body.encode('utf-8')
+    subject = subject.encode('utf-8')
+
+    queue = queue or current_app.config.get("BIBCATALOG_QUEUES")
+
+    payload = dict(
+        Queue=queue,
+        Subject=subject,
+        Text=body,
+        **kwargs
+    )
+
+    if recid:
+        payload["CF_RecordID"] = str(recid)
+
+    # Check if requests is set and also ignore admin due to RT mail loop
+    if requestors and "admin@inspirehep.net" not in requestors:
+        payload["requestors"] = requestors
+
+    return rt_instance.create_ticket(**payload)
+
+
+def create_ticket_with_template(queue,
+                                requestors,
+                                template_path,
+                                template_context,
+                                recid=None,
+                                **kwargs):
+    """Creates new RT ticket with a body that is rendered template
+
+    :param queue: where the ticket will be created
+    :type queue: string
+
+    :param requestors: username to set to requestors field of the ticket
+    :type requestors: string
+
+    :param template_path: path to the template for the ticket body
+
+    :param template_context: context object to be used to render template
+
+    :param recid: record id to be set custom RecordID field
+    :type recid: integer
+
+    :param kwargs: Other arguments possible to set:
+
+                Cc, AdminCc, Owner, Status,Priority, InitialPriority,
+                FinalPriority, TimeEstimated, Starts, Due,
+                ... (according to RT fields)
+
+                Custom fields CF.{<CustomFieldName>} could be set
+                with keywords CF_CustomFieldName.
+    :returns: ID of the new ticket or ``-1``, if it fails
+    :rtype: integer
+    """
+    body = render_template(template_path, **template_context).strip()
+    subject = template_context.get("subject")
```

Comment:
utf-8 encoding done in the fuction that's called after.

If you think it would be better, I could make it as a separate param, I didn't mind because it's already in the context.